release/models/gnpsi/openconfig-gnpsi.yang

```diff
@@ -0,0 +1,70 @@
+module openconfig-gnpsi {
+  yang-version "1";
+  namespace "http://openconfig.net/yang/gnpsi";
+  prefix "oc-gnpsi";
+
+  import openconfig-extensions {
+    prefix oc-ext;
+  }
+
+  import openconfig-system {
+    prefix oc-sys;
+  }
+
+  import openconfig-system-grpc {
+    prefix oc-sys-grpc;
+  }
+
+  import openconfig-gnpsi-types {
+    prefix oc-gnpsit;
+  }
+
+  import openconfig-yang-types {
+    prefix oc-yang;
+  }
+
+  organization
+    "OpenConfig Working group";
+  contact
+    "www.openconfig.net";
+
+  description
+    "This module defines configuration and operational state data
+    related to gRPC Network Packet Sampling Interface (gNPSI).
+
+    GNPSI Proto Reference: https://github.com/openconfig/gnpsi";
+
+  oc-ext:openconfig-version "0.1.0";
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  revision 2024-05-16 {
+    description
+      "Initial revision.";
+    reference "0.1.0";
+  }
+
+  grouping gnpsi-counters-top {
+    description
+      "Top-level container of operational data for gNPSI counters.";
+    leaf flow-send-error {
+      type oc-yang:counter64;
+      description
+        "A count of errors the gnpsi server encountered when sending a
+        flow message to a grpc client.  In sflow this is an sflow datagram.
```

Comment:
I'm still a bit puzzled by this.

If this is a generic error encountered when attempting to send a packet via a gRPC stream, why is it gNPSI specific?

Moreover, in the lemming reference that @sallylsy provided, the stream is closed immediately after encountering an error. Which, in many cases, will lead to the closure of the connection as well (and hence this data will not be available in telemetry).

Could you please elaborate on the purpose of this counter, and are there any real implementations of it?

Reply:
I discussed with Sally and gnpsi team.  It does seem like the gnpsi error sending flow message may be generalized as a error sending any grpc data.  I think it's up to the server implementation to decide if a send() error should be retried or the connection is closed.  But incrementing a counter that an error on send() occurred seems applicable to any grpc service.  The latest commit updates this.